Adding list of FLPs to GRPECS

chiarazampolli opened the pull request:
@shahor02 , @davidrohr , @ktf

chiarazampolli committed:
Addingl ist of FLPs to GRPECS

shahor02 reviewed (commented):
Hi @chiarazampolli 
not sure the bitset (change of N is not supported by schema evolution) is appropriate: what if in the future the range of IDs changes? I assume the ECS will provide a string of FLP hostnames, would store it directly.

davidrohr commented:
Hi, not sure if we want hostnames? Then we do need to do a lot of string comparison. How about a vector of integers with the flp numbers?

chiarazampolli commented:
I assumed that we will never have more than the present 202 FLPs. I would also not store the hostnames, and I hope FLP does not provide that. I can change it to a vector of integers, fine with me. The way we will fill it depends on what AliECS will pass.

shahor02 commented:
vector<int> would also work, provided the hostname format is carved in stone. The string comparisons would need to be done only once, at initialization.

shahor02 commented:
> I assumed that we will never have more than the present 202 FLPs.

The number of FLPs will probably not increase, but the IDs, theoretically, may.

davidrohr commented:
I don't see why we depend on the hostname format. Each hostname should contain the FLP number in one or the other form, so if we would just extract that once at the beginning, aliecs can also just put the numbers in directly. In case we would add more FLPs, it would work transparently, as long as they get a number.
Only problem would be if the numbering would change, i.e. TPC would suddenly have other FLP numbers. Then we'd need to ajust the code to connect to the correct FLPs.

shahor02 commented:
@davidrohr if you are using integers, don't you need to compose the hostname? Is it guaranteed that names will be always `alio2-cr1-flp<XXX>` ?

davidrohr commented:
Ok, point, that is what you meant. But also, we need to filter the host names to find zhe tpc flps to connect to. So I think we need the host name scheme in the source code anyway. Or we filter for just all numbers in the string, but don't really like that.

Kind Regards
David Rohr

Sent from my mobile. (Excuse the typos!)

On 24 August 2022 18:27:18 CEST, Ruben Shahoyan ***@***.***> wrote:
***@***.*** if you are using integers, don't you need to compose the hostname? Is it guaranteed that names will be always `alio2-cr1-flp<XXX>` ?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/AliceO2Group/AliceO2/pull/9690#issuecomment-1225956596
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

chiarazampolli commented:
We can also have vector of strings, but this will not help in knowing which FLP is for which detector. Such mapping should not be hardcoded, if you really think that it can change.
Additionally, remember that the GRP won’t be used for IDCs, but just in case this info is useful in reco.

chiarazampolli committed:
using vector<unsigned short>

shahor02 commented on DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h:
```diff
@@ -109,6 +110,18 @@ class GRPECSObject
   DetID::mask_t getDetsReadOut(const std::string& only, const std::string& skip = "") const { return getDetsReadOut(DetID::getMask(only), DetID::getMask(skip)); }
 
+  // methods to manipulate the list of FLPs in the run
+  std::vector<unsigned short> getListOfFLPs() const { return mFLPs; }
```
should be 
`const std::vector<unsigned short>& getListOfFLPs() const { return mFLPs; }`

chiarazampolli replied:
Sure, will change it ASAP.

chiarazampolli replied:
done.

shahor02 reviewed (changes requested):
thanks, just a small comment

chiarazampolli commented:
Hello @shahor02 , 
Can this be merged now? Or are there additional comments?
Thanks!
Chiara

shahor02 commented on DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h:
```diff
@@ -109,6 +110,18 @@ class GRPECSObject
   DetID::mask_t getDetsReadOut(const std::string& only, const std::string& skip = "") const { return getDetsReadOut(DetID::getMask(only), DetID::getMask(skip)); }
 
+  // methods to manipulate the list of FLPs in the run
+  const std::vector<unsigned short> getListOfFLPs() const { return mFLPs; }
```
please return `const std::vector<unsigned short>&`

shahor02 commented on DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h:
```diff
@@ -19,6 +19,7 @@
 #include <Rtypes.h>
 #include <cstdint>
 #include <ctime>
+#include <bitset>
```
not needed

shahor02 reviewed (changes requested):
sorry, forgot about this PR, see below

chiarazampolli commented:
Done.

chiarazampolli commented:
Maybe at this point we can add here the change to the executable too. So let's wait to merge. I will take care.

chiarazampolli committed:
List of FLPs filled with a comma-separated list

alibuild commented:
Error while checking build/O2/o2-dataflow for 4d95e1431160d39b244ecd929d695a561b6923c4 at 2022-09-02 16:53:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 423
 88/106 Test #102: test_Framework_test_SimpleDataProcessingDevice01 ........***Failed    2.85 sec
[ERROR] pid 24559 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (24559) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/4d95e1431160d39b244ecd929d695a561b6923c4/build_O2_o2-dataflow/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow for 3740a92a1bfb10769467e0fe0325947b3f155d2d at 2022-09-02 16:57:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc7_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:232:30: error: invalid initialization of reference of type 'const string&' {aka 'const std::__cxx11::basic_string<char>&'} from expression of type 'const long int'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/3740a92a1bfb10769467e0fe0325947b3f155d2d/build_O2_o2-dataflow/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow-cs8 for 3740a92a1bfb10769467e0fe0325947b3f155d2d at 2022-09-02 17:31:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:232:30: error: invalid initialization of reference of type 'const string&' {aka 'const std::__cxx11::basic_string<char>&'} from expression of type 'const long int'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/3740a92a1bfb10769467e0fe0325947b3f155d2d/build_O2_o2-dataflow-cs8/pretty.html).

alibuild commented:
Error while checking build/O2/o2 for eb84954ecc10efd7ab9828376e02ebc5153e9123 at 2022-09-03 04:06:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 446
 95/106 Test #109: test_Framework_test_SingleDataSource ....................***Failed    2.86 sec
[ERROR] pid 26685 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (26685) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/eb84954ecc10efd7ab9828376e02ebc5153e9123/build_O2_o2/pretty.html).

chiarazampolli commented:
Hello @TimoWilken , @ktf ,
The error does not point to this PR, can you check?
```
[INFO] 1st reading of A: testObjectO
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingB/1000/febcfe9c-2add-11ed-a816-c0a80209250c for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
[INFO] Reading of cached and modified A: Cached
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/CachingA/500/
[INFO] Read for wrong time, expect null: 0
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingA/1000/febc01dc-2add-11ed-a816-c0a80209250c for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
[INFO] Reading of A from scratch after error: testObjectO
--
[INFO] Reading same A, expect cached and modified value: Cached
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingB/1000/fee59254-2add-11ed-a816-c0a80209250c for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
[INFO] Reading B, expect cached since only A cache was cleaned: testObjectO
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testBasicCCDBManager.cxx(97): [1;31;49merror: in "TestBasicCCDBManager": check objB && (*objB) == hack has failed[0;39;49m
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingB/1000/fee59254-2add-11ed-a816-c0a80209250c for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
[INFO] Reading B after cleaning cache completely: testObjectO
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/CachingA/1500/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/CachingA/1500/
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingA for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
[INFO] Reading A after disabling the cache: testObjectO
[INFO] ccdb reads http://ccdb-test.cern.ch:8080/Test/CachingA for 1500 (retrieve, agent_id: aliflow01.cern.ch-1662136846-teP1Bu), 
--
[26336:simple]: [16:49:50][INFO] Correctly handshaken websocket connection.
[INFO] Quitting
[INFO] QUIT_REQUESTED
[ERROR] pid 26337 (internal-dpl-injected-dummy-sink) crashed with 1
[26335:internal-dpl-clock]: [16:49:50][INFO] Received device shutdown request (signal 15).
[26335:internal-dpl-clock]: [16:49:50][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[26335:internal-dpl-clock]: [16:49:50][INFO] New state requested. No timeout set, quitting immediately as per --completion-policy
--
[26336:simple]: [16:49:51][STATE] IDLE ---> EXITING
[26336:simple]: [16:49:51][STATE] Exiting FairMQ state machine
[INFO] Dumping used configuration in dpl-config.json
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (26337) returned with 1

        Start 103: test_Framework_test_SimpleRDataFrameProcessing
 89/106 Test #103: test_Framework_test_SimpleRDataFrameProcessing ..........   Passed    7.10 sec
--
```

shahor02 committed:
pass vectors by reference

shahor02 committed:
use tokenize and safer std::stoi for flps IDs extraction

shahor02 committed:
formatting

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS for 9080af322fe9b008c5aec60d774217b37bcaddba at 2022-09-03 17:52:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/nomad/alloc/f7552d80-4750-bbd0-a1cd-d8b3785f2966/ci/local/o2/sw/SOURCES/O2/9690/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:59:7: error: use of undeclared identifier 'addFlp'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/9080af322fe9b008c5aec60d774217b37bcaddba/build_AliceO2_O2_o2_macOS/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow for 9080af322fe9b008c5aec60d774217b37bcaddba at 2022-09-03 17:59:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc7_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:59:7: error: 'addFlp' was not declared in this scope
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/9080af322fe9b008c5aec60d774217b37bcaddba/build_O2_o2-dataflow/pretty.html).

shahor02 committed:
Update create-grp-ecs.cxx

alibuild commented:
Error while checking build/O2/o2-dataflow-cs8 for 10555155d380cb643a4feaf1cde6f6a98974f95b at 2022-09-03 18:17:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:59:7: error: 'addFlp' was not declared in this scope
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/10555155d380cb643a4feaf1cde6f6a98974f95b/build_O2_o2-dataflow-cs8/pretty.html).

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS-arm for 10555155d380cb643a4feaf1cde6f6a98974f95b at 2022-09-03 18:33:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9690/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:59:7: error: use of undeclared identifier 'addFlp'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/10555155d380cb643a4feaf1cde6f6a98974f95b/build_AliceO2_O2_o2_macOS-arm/pretty.html).

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS-arm for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 18:34:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9690/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: no member named 'addFlp' in 'o2::parameters::GRPECSObject'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_AliceO2_O2_o2_macOS-arm/pretty.html).

alibuild commented:
Error while checking build/O2/o2-cs8 for 10555155d380cb643a4feaf1cde6f6a98974f95b at 2022-09-03 18:36:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:59:7: error: 'addFlp' was not declared in this scope
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/10555155d380cb643a4feaf1cde6f6a98974f95b/build_O2_o2-cs8/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 18:37:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc7_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: 'using GRPECSObject = class o2::parameters::GRPECSObject' {aka 'class o2::parameters::GRPECSObject'} has no member named 'addFlp'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_O2_o2-dataflow/pretty.html).

alibuild commented:
Error while checking build/AliceO2/O2/o2/macOS for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 19:07:
```
## sw/BUILD/O2-latest/log
/System/Volumes/Data/build/nomad/alloc/f7552d80-4750-bbd0-a1cd-d8b3785f2966/ci/local/o2/sw/SOURCES/O2/9690/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: no member named 'addFlp' in 'o2::parameters::GRPECSObject'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_AliceO2_O2_o2_macOS/pretty.html).

alibuild commented:
Error while checking build/O2/o2 for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 19:07:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc7_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: 'using GRPECSObject = class o2::parameters::GRPECSObject' {aka 'class o2::parameters::GRPECSObject'} has no member named 'addFlp'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_O2_o2/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow-cs8 for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 19:19:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: 'using GRPECSObject = class o2::parameters::GRPECSObject' {aka 'class o2::parameters::GRPECSObject'} has no member named 'addFlp'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_O2_o2-dataflow-cs8/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-06 20:00:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: 'using GRPECSObject = class o2::parameters::GRPECSObject' {aka 'class o2::parameters::GRPECSObject'} has no member named 'addFlp'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/o2-cs8 for 0592d9067b3932072e9604353e3a70734116cba4 at 2022-09-03 20:15:
```
## sw/BUILD/O2-latest/log
/sw/SOURCES/O2/9690-slc8_x86-64/0/Detectors/GRP/workflows/src/create-grp-ecs.cxx:80:14: error: 'using GRPECSObject = class o2::parameters::GRPECSObject' {aka 'class o2::parameters::GRPECSObject'} has no member named 'addFlp'; did you mean 'addFLP'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/0592d9067b3932072e9604353e3a70734116cba4/build_O2_o2-cs8/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow-cs8 for b644d1f0df784318d7f8c1684cdb48d72398f173 at 2022-09-03 21:14:
```
## sw/BUILD/O2-latest/log
122/423 Test  #91: CCDB/test/testCcdbApi_ConfigParam.cxx ................................................***Failed   32.02 sec
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/GLO/Config/PVertexer/1662231737831/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi_ConfigParam.cxx(130): [1;31;49merror: in "testConfigParamRetrieval": check object != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi_ConfigParam.cxx(132): [1;31;49merror: in "testConfigParamRetrieval": check object->getMemberProvenance("useMeanVertexConstraint") == o2::conf::ConfigurableParam::EParamProvenance::kCCDB has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi_ConfigParam.cxx(134): [1;31;49merror: in "testConfigParamRetrieval": check p1.getMemberProvenance("useMeanVertexConstraint") == o2::conf::ConfigurableParam::EParamProvenance::kCCDB has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi_ConfigParam.cxx(135): [1;31;49merror: in "testConfigParamRetrieval": check object == &p1 has failed[0;39;49m
209/423 Test  #93: CCDB/test/testBasicCCDBManager.cxx ...................................................***Failed  123.98 sec
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/CachingA/1500/
[FATAL] Got nullptr from CCDB for path Test/CachingA and timestamp 1500
unknown location(0): [4;31;49mfatal error: in "TestBasicCCDBManager": unknown type[0;39;49m
423/423 Test  #90: CCDB/test/testCcdbApi.cxx ............................................................***Failed  577.12 sec
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot insert the object in the database</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPath/1662231842989/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(142): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 != nullptr has failed [0 == nullptr][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(145): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 && path2->getPathString().CompareTo("HelloWorld") == 0 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPath/1662231859658/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPath/1662231873292/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(153): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md.count("Hello") == 1 has failed [0 != 1][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(154): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md["Hello"] == "World" has failed [ != World][0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot insert the object in the database</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/tree2/1662231932898/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(168): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(169): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(170): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870//1662231937418/
[ERROR] Unable to find object Test/pid15870//1662231937418, Aborting
[ERROR] Local snapshot /tmp/A8tPUgnOzdIafz94/Test/pid15870/CCDBPath/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(188): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check snapshot.retrieveFromTFileAny<o2::ccdb::IdPath>(basePath + "CCDBPath", f.metadata) != nullptr has failed[0;39;49m
[ERROR] Local snapshot /tmp/A8tPUgnOzdIafz94/Test/pid15870/tree2/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(192): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(193): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(194): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot insert the object in the database</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(237): [1;31;49merror: in "timestamptest": check path2 != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1001/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1002/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1003/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1004/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1005/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1006/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1007/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15870/CCDBPathUnitTest/1008/Hello=World/
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/b644d1f0df784318d7f8c1684cdb48d72398f173/build_O2_o2-dataflow-cs8/pretty.html).

chiarazampolli commented:
Hello @shahor02 , 
Can this be merged?
Chiara

shahor02 committed:
fix typo

shahor02 commented:
actually, all CIs were failed due to the typo (perhaps in my fix)

alibuild commented:
Error while checking build/O2/o2-dataflow for f2fb0ab249760774a8b7328e3f9721ac4a220f05 at 2022-09-10 08:45:
```
## sw/BUILD/O2-latest/log
424/424 Test  #90: CCDB/test/testCcdbApi.cxx ............................................................***Failed    1.69 sec
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;81&#47;4012&#47;eaa08d88-30d3-11ed-8a63-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPath/1662792225095/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(142): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 != nullptr has failed [0 == nullptr][0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(145): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 && path2->getPathString().CompareTo("HelloWorld") == 0 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPath/1662792225098/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPath/1662792225101/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(153): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md.count("Hello") == 1 has failed [0 != 1][0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(154): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md["Hello"] == "World" has failed [ != World][0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;25&#47;7795&#47;eaa6cb0c-30d3-11ed-8a63-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/tree2/1662792225137/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(168): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(169): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(170): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619//1662792225140/
[ERROR] Unable to find object Test/pid15619//1662792225140, Aborting
[ERROR] Local snapshot /tmp/IcD3XmNW3UgXdunA/Test/pid15619/CCDBPath/snapshot.root not found 
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(188): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check snapshot.retrieveFromTFileAny<o2::ccdb::IdPath>(basePath + "CCDBPath", f.metadata) != nullptr has failed[0;39;49m
[ERROR] Local snapshot /tmp/IcD3XmNW3UgXdunA/Test/pid15619/tree2/snapshot.root not found 
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(192): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(193): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(194): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;93&#47;9364&#47;eaab9664-30d3-11ed-8a63-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(237): [1;31;49merror: in "timestamptest": check path2 != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1001/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1002/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1003/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1004/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1005/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1006/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1007/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1008/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1009/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/999/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1010/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/CCDBPathUnitTest/1662792225206/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(270): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers.count("Hello") == 1 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(271): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers["Hello"] == "World" has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15619/th1/1662792225214/Hello=World/
/sw/SOURCES/O2/9690-slc7_x86-64/0/CCDB/test/testCcdbApi.cxx(279): [1;31;49merror: in "retrieveTMemFile_test": check obj != nullptr has failed [0 == nullptr][0;39;49m
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/f2fb0ab249760774a8b7328e3f9721ac4a220f05/build_O2_o2-dataflow/pretty.html).

alibuild commented:
Error while checking build/O2/o2-dataflow-cs8 for f2fb0ab249760774a8b7328e3f9721ac4a220f05 at 2022-09-10 09:32:
```
## sw/BUILD/O2-latest/log
114/424 Test  #90: CCDB/test/testCcdbApi.cxx ............................................................***Failed    9.87 sec
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;59&#47;4950&#47;ed81a95a-30d9-11ed-8a65-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPath/1662794806907/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(142): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 != nullptr has failed [0 == nullptr][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(145): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 && path2->getPathString().CompareTo("HelloWorld") == 0 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPath/1662794806909/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPath/1662794806911/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(153): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md.count("Hello") == 1 has failed [0 != 1][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(154): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md["Hello"] == "World" has failed [ != World][0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;66&#47;4092&#47;ed86f9bb-30d9-11ed-8a65-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/tree2/1662794806942/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(168): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(169): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(170): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880//1662794806945/
[ERROR] Unable to find object Test/pid15880//1662794806945, Aborting
[ERROR] Local snapshot /tmp/Pe8U4BUvl5QrlGdk/Test/pid15880/CCDBPath/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(188): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check snapshot.retrieveFromTFileAny<o2::ccdb::IdPath>(basePath + "CCDBPath", f.metadata) != nullptr has failed[0;39;49m
[ERROR] Local snapshot /tmp/Pe8U4BUvl5QrlGdk/Test/pid15880/tree2/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(192): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(193): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(194): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;65&#47;2367&#47;ed8a5a14-30d9-11ed-8a65-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(237): [1;31;49merror: in "timestamptest": check path2 != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1001/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1002/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1003/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1004/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1005/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1006/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1007/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1008/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1009/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/999/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1010/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/CCDBPathUnitTest/1662794806990/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(270): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers.count("Hello") == 1 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(271): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers["Hello"] == "World" has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid15880/th1/1662794806995/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(279): [1;31;49merror: in "retrieveTMemFile_test": check obj != nullptr has failed [0 == nullptr][0;39;49m
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/f2fb0ab249760774a8b7328e3f9721ac4a220f05/build_O2_o2-dataflow-cs8/pretty.html).

chiarazampolli commented:
Hello @shahor02 ,

now it is really unrelated, the error. I don’t know what is happening with CCDB, but I would consider merging this.

chiara

alibuild commented:
Error while checking build/O2/o2-cs8 for f2fb0ab249760774a8b7328e3f9721ac4a220f05 at 2022-09-10 08:00:
```
## sw/BUILD/O2-latest/log
123/447 Test  #90: CCDB/test/testCcdbApi.cxx ............................................................***Failed   24.51 sec
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;19&#47;6054&#47;6c58d89c-30cc-11ed-8a60-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPath/1662789006755/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(142): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 != nullptr has failed [0 == nullptr][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(145): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check path2 && path2->getPathString().CompareTo("HelloWorld") == 0 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPath/1662789006758/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPath/1662789006760/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(153): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md.count("Hello") == 1 has failed [0 != 1][0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(154): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check md["Hello"] == "World" has failed [ != World][0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;20&#47;6842&#47;6c5e0d4d-30cc-11ed-8a60-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/tree2/1662789006790/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(168): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(169): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(170): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671//1662789006793/
[ERROR] Unable to find object Test/pid16671//1662789006793, Aborting
[ERROR] Local snapshot /tmp/UrnZwd25GV05NJjv/Test/pid16671/CCDBPath/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(188): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check snapshot.retrieveFromTFileAny<o2::ccdb::IdPath>(basePath + "CCDBPath", f.metadata) != nullptr has failed[0;39;49m
[ERROR] Local snapshot /tmp/UrnZwd25GV05NJjv/Test/pid16671/tree2/snapshot.root not found 
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(192): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(193): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && std::strcmp(tree->GetName(), "tree123") == 0 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(194): [1;31;49merror: in "store_retrieve_TMemFile_templated_test": check tree != nullptr && tree->GetEntries() == 1 has failed[0;39;49m
<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Cannot upload the blob to the local file &#47;root&#47;QC&#47;10&#47;3051&#47;6c614ca3-30cc-11ed-8a60-c0a80209250c</p><p><b>Description</b> The server encountered an unexpected condition that prevented it from fulfilling the request.</p><hr class="line" /></body></html>[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(237): [1;31;49merror: in "timestamptest": check path2 != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1000/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1001/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1002/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1003/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1004/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1005/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1006/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1007/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1008/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1009/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(242): [1;31;49merror: in "timestamptest": check p != nullptr has failed [0 == nullptr][0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/999/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1010/Hello=World/
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/CCDBPathUnitTest/1662789006841/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(270): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers.count("Hello") == 1 has failed[0;39;49m
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(271): [1;31;49merror: in "retrieveTemplatedWithHeaders": check headers["Hello"] == "World" has failed[0;39;49m
[ERROR] Requested resource does not exist: http://ccdb-test.cern.ch:8080/Test/pid16671/th1/1662789006845/Hello=World/
/sw/SOURCES/O2/9690-slc8_x86-64/0/CCDB/test/testCcdbApi.cxx(279): [1;31;49merror: in "retrieveTMemFile_test": check obj != nullptr has failed [0 == nullptr][0;39;49m
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9690/f2fb0ab249760774a8b7328e3f9721ac4a220f05/build_O2_o2-cs8/pretty.html).
